components/observations/ObservationsFilterForm.tsx

```diff
@@ -42,30 +42,30 @@ const observationFilterConfigSchema = z
   .deepPartial();
 export type ObservationFilterConfig = z.infer<typeof observationFilterConfigSchema>;
 
-const matchesDates = (dates: z.infer<typeof observationFilterConfigSchema.shape.dates>, observation: ObservationFragment): boolean => {
+const matchesDates = (currentDate: Date, dates: z.infer<typeof observationFilterConfigSchema.shape.dates>, observation: ObservationFragment): boolean => {
   let matches = true;
   if (!dates) {
     return matches;
   }
   switch (dates.value) {
     case 'past_day':
-      dates.to = new Date();
```

Comment:
using `new Date()` here breaks date filtering when we're pinned to a non-default date